core/exception/name_spec.rb

```diff
@@ -1,5 +1,35 @@
 require File.expand_path('../../../spec_helper', __FILE__)
 
 describe "NameError#name" do
-  it "needs to be reviewed for spec completeness"
+  it "returns a method name as a symbol" do
+    begin
+      doesnt_exist
+    rescue NameError => error
+      error.name.should == :doesnt_exist
+    end
```

Comment:
This should use the `raise_error` matcher with a block:
```ruby
lambda {
  doesnt_exist
}.should raise_error(NameError) { |e| e.name.should == :doesnt_exist }
```

Otherwise the spec might fail with a "no assertions error" which is less clear.